modules/default/weather/providers/openweathermap.js

```diff
@@ -95,6 +116,18 @@ WeatherProvider.register("openweathermap", {
 		return days;
 	},
 
+	/*
+	 * Generate WeatherObjects based on One Call forecast information
+	 */
+	generateWeatherObjectsFromOnecall(data) {
+		if (this.config.weatherEndpoint === "/onecall") {
+			return this.fetchOnecall(data);
+		}
+		// if weatherEndpoint does not match onecall, what should be returned?
```

Comment:
is that a real question or a rhetorical? is there a TODO still here?

Reply:
Rhetorical, the `generateWeatherObjectsFromForecast()` method I based the new one off of had a similar comment, I guess because no one settled on what type of "empty" object the method should return.